plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/internal/ProjectSelectorSelectionChangedListener.java

```diff
@@ -0,0 +1,75 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.deploy.ui.internal;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.cloud.tools.eclipse.appengine.deploy.ui.Messages;
+import com.google.cloud.tools.eclipse.login.ui.AccountSelector;
+import com.google.cloud.tools.eclipse.projectselector.GcpProject;
+import com.google.cloud.tools.eclipse.projectselector.ProjectRepository;
+import com.google.cloud.tools.eclipse.projectselector.ProjectRepositoryException;
+import com.google.cloud.tools.eclipse.projectselector.ProjectSelector;
+import java.text.MessageFormat;
+import org.eclipse.jface.viewers.ISelectionChangedListener;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.SelectionChangedEvent;
+
+public class ProjectSelectorSelectionChangedListener implements ISelectionChangedListener {
+
+  private static String CREATE_APP_LINK =
+      "https://console.cloud.google.com/appengine/create?lang=java&project={0}";
+
+  private final AccountSelector accountSelector;
+  private final ProjectRepository projectRepository;
+  private final ProjectSelector projectSelector;
+
+  public ProjectSelectorSelectionChangedListener(AccountSelector accountSelector,
+                                                 ProjectRepository projectRepository,
+                                                 ProjectSelector projectSelector) {
+    this.accountSelector = accountSelector;
+    this.projectRepository = projectRepository;
+    this.projectSelector = projectSelector;
+  }
+
+  @Override
+  public void selectionChanged(SelectionChangedEvent event) {
+    IStructuredSelection selection = (IStructuredSelection) event.getSelection();
+    try {
+      if (!selection.isEmpty()) {
+        GcpProject project = (GcpProject) selection.getFirstElement();
+        Credential selectedCredential = accountSelector.getSelectedCredential();
+        String projectId = project.getId();
+        boolean hasAppEngineApplication =
+            projectRepository.hasAppEngineApplication(selectedCredential, projectId);
+        if (!hasAppEngineApplication) {
+          String link = MessageFormat.format(CREATE_APP_LINK, projectId);
+          projectSelector.setStatusLink(
+              Messages.getString("projectselector.missing.appengine.application.link",
+                                 link), link);
+        } else {
+          projectSelector.clearStatusLink();
+        }
+      } else {
+        projectSelector.clearStatusLink();
+      }
+    } catch (ProjectRepositoryException ex) {
+      projectSelector.setStatusLink(Messages.getString("projectselector.retrieveapplication.error.message",
```

Comment:
If the call has failed, will this prevent us from deploying or do we assume deploy can go forward anyway?